var/spack/repos/builtin/packages/py-aiofiles/package.py

```diff
@@ -13,7 +13,9 @@ class PyAiofiles(PythonPackage):
     homepage = "https://github.com/Tinche/aiofiles"
     pypi = "aiofiles/aiofiles-0.5.0.tar.gz"
 
+    version("0.7.0", sha256="a1c4fc9b2ff81568c83e21392a82f344ea9d23da906e4f6a52662764545e19d4")
     version("0.5.0", sha256="98e6bcfd1b50f97db4980e182ddd509b7cc35909e903a8fe50d8849e02d815af")
 
     depends_on("py-setuptools", type="build")
     depends_on("py-wheel", type="build")
+    depends_on("py-poetry-core@1:", type="build", when="@0.7.0")
```

Comment:
`py-setuptools` is not needed anymore starting from `@0.7:` and `py-wheel` is automatically added by the `PythonPackage` class
`py-poetry-core` is also used in future releases (see version 0.8.0)
```suggestion
    depends_on("py-poetry-core@1:", when="@0.7:", type="build")

    # Historical dependencies
    depends_on("py-setuptools", when="@:0.6", type="build")
```